tests/mail/tests.py

```diff
@@ -53,17 +53,36 @@ def assertMessageHasHeaders(self, message, headers):
         if isinstance(message, bytes):
             message = message_from_bytes(message)
         msg_headers = set(message.items())
-        self.assertTrue(
-            headers.issubset(msg_headers),
-            msg="Message is missing "
-            "the following headers: %s" % (headers - msg_headers),
-        )
+        if not headers.issubset(msg_headers):
+            missing = "\n".join(f"  {h}: {v}" for h, v in headers - msg_headers)
+            actual = "\n".join(f"  {h}: {v}" for h, v in msg_headers)
+            raise self.failureException(
+                f"Expected headers not found in message.\n"
+                f"Missing headers:\n{missing}\n"
+                f"Actual headers:\n{actual}"
+            )
 
+    def assertStartsWith(self, first, second):
+        if not first.startswith(second):
+            # Show beginning of first to see difference in failure message.
+            self.longMessage = True
+            start_of_first = first[: len(second) + 15]
```

Comment:
The `+ 15` is new (also in `assertEndsWith`) and I'm not sure why it's needed :thinking: 

Everything else in this commit (0891625557d2da2e87749bba49f57639c2a83110) makes sense to me :+1:

Reply:
Fair point. I just pushed an improved version (which I'm hoping also explains itself better).